### DISCUSSION TITLE
Update H2GIS to 2.0.0

### DISCUSSION THREAD

**Discussion by ebocher** [dbeaver/dbeaver#15379] [DBeaver Development]
H2GIS 2.0.0 is out on Maven Central.
We have updated it on your plugin repo https://github.com/orbisgis/orbisrcp/tree/master/plugins/org.orbisgis.h2gis
But we have an issue to detect a Geometry data type.
DBeaver set the type name from a plugin.xml file, here https://github.com/orbisgis/orbisrcp/tree/master/plugins/org.orbisgis.h2gis
Since H2 2.X version the string type representation for a geometry has changed. See my issue to the H2 community https://github.com/h2database/h2database/issues/3419
Is there a way to set the type name to DBeaver from code and not from a static value in a file ?
Best regards

**Comment by ShadelessFox** [comment]
Hello @ebocher,

If I understand you correctly, you can use `<type name="*"/>` instead of specifying a concrete type name.

**Reply by ebocher** [reply]
Not sure if I must used ```<type name="*"/>``` because each geometry column can return different type name value depending on the SRID value. 
We can have : 
```GEOMETRY
GEOMETRY(POINT)
GEOMETRY(POINT, 4326) //where 4326 is the SRID value
```
instead of GEOMETRY in the old H2GIS version.

**Reply by ShadelessFox** [reply]
You can use `startsWith` inside your value handler provider to determine if it's geometry type or not.

Maybe it is possible to use the type id instead? Perhaps geometry type has a unique id beyond the ones defined in `java.sql.Types`.

**Reply by ebocher** [reply]
Thanks for the answer.
geometry is not defined as a java.sql.Types
I will change here https://github.com/orbisgis/orbisrcp/blob/master/plugins/org.orbisgis.h2gis/src/org/orbisgis/h2gis/data/H2GIS200ValueHandlerProvider.java#L35 the value handler

**Comment by ebocher** [comment]
Just to inform you that H2GIS 2.0 rocks  with DBeaver  with your proposal.
We are waiting now a fix on https://github.com/dbeaver/dbeaver/issues/15556 to send a PR to dbeaver.
A working version of H2GIS is only available here  https://github.com/orbisgis/orbisrcp/tree/master/plugins/org.orbisgis.h2gis
Thanks
